bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/services/UrlProviderImpl.java

```diff
@@ -416,12 +493,22 @@ private String getIdentifierFromRequest(SlingHttpServletRequest request) {
     }
 
     @Override
-    public ProductUrlFormat.Params parseProductUrlFormatParameters(SlingHttpServletRequest request) {
-        return newProductUrlFormat.parse(request.getRequestPathInfo(), request.getRequestParameterMap());
+    public ProductUrlFormat.Params parseProductUrlFormatParameters(SlingHttpServletRequest request, Page page) {
```

Comment:
You don't need the `Page` here. Please obtain the context from the request. There are multiple ways you could do that, 

a) get the `currentPage` from the wcm bindings
b) resolve the original requested resource again
c) or just use the request resource